File: 3 - Proyecto/Despacho/Datos/Modelo/EmpresaTransporte.cs
using System.Data;

namespace Datos.Modelo
{
	public class EmpresaTransporte: Base.ModeloBase
	{
		public int EmpresaTransporteId { get; set; }
		public string Nombre { get; set; }
		public bool EsPropia { get; set; }
		public bool EstaActivo { get; set; }
		public void FromDataRow(DataRow fila)
		{
			this.EmpresaTransporteId = int.Parse(fila[0].ToString());
			this.Nombre = fila[1].ToString();
			this.EsPropia = bool.Parse(fila[2].ToString());
			this.EstaActivo = bool.Parse(fila[3].ToString());
		}
	}
}

File: 3 - Proyecto/Deploy/Content/js/EmpresaTransporte.js
let EmpresaTransporte = function () {

  let Init = function () {
    InitElementos();
  };

  let InitElementos = function () {
    if ($("#listaempresastransporte").length > 0) {
      cargarTabla("EmpresaTransporteId", "EmpresaTransporte", { Id: 0 }, "#listaempresatransporte", "#buscarempresatransporte", [
        { field: "EmpresaTransporteId", title: "#", width: 50, selector: !1, textAlign: "center" },
        { field: "Nombre", title: "Nombre", responsive: { visible: "lg" } },
        { field: "EsPropia", title: "Es Propia", responsive: { visible: "lg" }, template: function (e, a, i) { return e.EsPropia == true ? "Si" : "No"; }  },
        { field: "EstaActivo", title: "Activo", responsive: { visible: "lg" }, template: function (e, a, i) { return e.EstaActivo == true ? "Si" : "No"; } }
      ], true, true);
    }

    $('#btnGuardar').click(function () {
      let id = $('#id').val();
      let activo = $('#activo').val();
      let nombre = $('#nombre').val();
      let espropia = $('#espropia').val();
      
      $.post("/EmpresaTransporte/" + (id > 0 ? "Edit" : "Create"), {
        EmpresaTransporteId: id,
        Nombre: nombre,
        EsPropia: espropia,
        EstaActivo: activo
      }, function (data) {
        if (data.exito) {
          mensaje("Éxito", "Información guardada correctamente", "exito", function () { location.href = "/EmpresaTransporte/Index"; });
        } else {
          mensaje("Error", "No se pudo guardar la información", "error");
        }
      });
    });
  };



  return {
    init: function () {
      Init();
    }
  };
}();

$(() => {
  EmpresaTransporte.init();
});